ezprefs.lua

```diff
@@ -0,0 +1,966 @@
+    print(text)
+    dfhack.color(-1)
+end
+
+
+
+----------------------- PROCESS_UNIT -------------------------------
+local function process_unit(unit)
+    local unitnum
+
+    if unit then
+        unitnum = tonumber(unit)
+
+        if (not unitnum) and opts.unit then
+            print_yellow("'" .. unit .. "' is not a valid unit ID.")
+            opts.action = "die"
+            return
+        end
+    else
+        local tempunit = dfhack.gui.getSelectedUnit(true)
+
+        if not tempunit then
+            print_yellow("No unit specified or selected with [v]iew in Dwarf Fortress. Aborting.")
+            opts.action = "die"
+            return
+        else
+            unit = tempunit.id
+        end
+
+        unitnum = tonumber(unit)
+    end
+
+    local unitobj = df.unit.find(unitnum)
+
+    if not unitobj then
+        print_yellow("Cannot find unit " .. unit .. ".")
+        opts.action = "die"
+        return
+    end
+
+    opts.unitname = dfhack.df2console(dfhack.TranslateName(dfhack.units.getVisibleName(unitobj)))
+    if not opts.silent then
+        print("Identified target " .. opts.unitname .. ".")
+    end
+    return unit
+end
+
+----- ------------------------------------------------- PRINT_ALIASES -------------------------------------------------- --
+local function print_aliases()
+    local aliases_formatted = {}
+
+    print("\nProfile names   Aliases")
+    print("=============   ====================================================================")
+
+    for _,v in ipairs(big_job_list) do
+        for j,w in pairs(job_alias[v]) do
+            if #w > 9 then
+                aliases_formatted[j] = w
+            else
+                aliases_formatted[j] = w .. string.rep(" ",9-#w)
+            end
+        end
+        print(table.concat(aliases_formatted,"\t"))
+        for k, _ in pairs(aliases_formatted) do
+            aliases_formatted[k]= nil
+        end
+    end
+    return
+end
+
+----------------------- GET_ALL_UNITS -------------------------------
+local function get_all_units()
+    local ret = {}
+    for _, thisunit in ipairs(df.global.world.units.all) do
+        if dfhack.units.isCitizen(thisunit) then
+            table.insert(ret,tostring(thisunit.id))
+        end
+    end
+    return ret
+end
+
+----------------------- VALIDATE_UNIT_OPTS -------------------------------
+local function validate_unit_opts()
+    if opts.unit then
+        if opts.all then
+            print_yellow("Option \'--unit\' conflicts with option \'--all\'.")
+            opts.action = "die"
+            return
+        elseif opts.preview then
+            print_yellow("Preview is hypothetical. Ignoring option \'--unit\'.")
+            opts.unit = false
+            return
+        else
+            process_unit(opts.unit)
+        end
+    elseif not opts.all then
+        if not opts.preview then
+            opts.unit = process_unit() -- looking for a selected unit
+            unit=opts.unit

```

Comment:
`unit` is a global var in this scope. are you missing a `local unit` in this function?